core/geometry/src/curve/Arc3d.ts

```diff
@@ -226,6 +226,35 @@ export class Arc3d extends CurvePrimitive implements BeJSONFunctions {
       result,
     );
   }
+  /**
+   * Create an arc from three points on the ellipse: two points on an axis and one in between.
+   * @param point0 start of arc, on an axis
+   * @param point1 point on arc somewhere between `point0` and `point2`
+   * @param point2 point on arc directly opposite `point0`
+   * @param sweep angular sweep, measured from `point0` in the direction of `point1`.
+   *  For a semicircle from `point0` to `point2` passing through `point1`, pass `AngleSweep.createStartEndDegrees(0,180)`.
+   *  Default value is full sweep to create the entire ellipse.
+   * @param result optional preallocated result
+   * @returns elliptical arc, or undefined if construction impossible.
+   */
+  public static createStartMiddleEnd(point0: XYAndZ, point1: XYAndZ, point2: XYAndZ, sweep?: AngleSweep, result?: Arc3d): Arc3d | undefined {
+    const center = Point3d.createAdd2Scaled(point0, 0.5, point2, 0.5);
+    const vector0 = Vector3d.createStartEnd(center, point0);
+    const vector1 = Vector3d.createStartEnd(center, point1);
+    const v0DotV1 = vector0.dotProduct(vector1);
+    const v0Len2 = vector0.magnitudeSquared();
+    if (Math.abs(v0DotV1) >= v0Len2)
+      return undefined; // middle point projects to end of axis or beyond (rules out negative under the radical below)
+    const normal = vector0.crossProduct(vector1);
+    const vector90 = normal.unitCrossProductWithDefault(vector0, 0, 0, 0);
+    const v1DotV90 = vector1.dotProduct(vector90);
+    // Solve the standard ellipse equation for the unknown axis length, given local coords of point1 (v0.v1/||v0||, v90.v1)
```

Comment:
I would mention `||v90|| = 1` at this point.
```suggestion
    // Solve the standard ellipse equation for the unknown axis length, given local coords of point1 (v0.v1/||v0||, v90.v1). Note that `||v90|| = 1` at this point.
```